packages/cli/src/commands/open.ts

```diff
@@ -1,20 +1,32 @@
 import { readConfig } from "@allurereport/core";
 import { serve } from "@allurereport/static-server";
 import { Command, Option } from "clipanion";
+import { glob } from "glob";
+import { join } from "node:path";
+import { cwd as processCwd } from "node:process";
+import { generate } from "./commons/generate.js";
 
 export class OpenCommand extends Command {
-  static paths = [["open"]];
+  static paths = [["open"], ["serve"]];
 
   static usage = Command.Usage({
     description: "Serves specified directory",
-    details: "This command serves the specified report directory and opens it in the default browser.",
+    details: "This command generates report with the given test results and opens it in the default browser.",
     examples: [
-      ["open", "Serve the default report directory"],
-      ["open custom-report --port 8080 --live", "Serve the custom-report directory on port 8080 with live reload"],
+      ["open", "Generate and serve the report based on default test results directories"],
+      ["open ./allure-results", "Generate and serve the report based on given test results directory"],
+      ["open --port 8080 ./allure-results", "Serve the report on port 8080"],
     ],
   });
 
-  reportDir = Option.String({ required: false, name: "The directory with Allure results" });
+  resultsDir = Option.String({
+    required: false,
+    name: "Pattern to match test results directories in the current working directory (default: ./**/allure-results)",
+  });
+
+  output = Option.String("--output,-o", {
+    description: "The output directory name. Absolute paths are accepted as well (default: allure-report)",
```

Comment:
We should generate into a temp directory by default:

  - this way, we won't overwrite an existing report
  - it better suits the idea of `allure serve` in Allure 2, which is "I don't care about the physical report, just open these allure results quickly`